api/tasks.go

```diff
@@ -58,27 +78,36 @@ func UpdateTask(c *gin.Context) {
 		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "incorrectly formatted time"})
 		return
 	}
-	status, err := strconv.Atoi(c.Query("status"))
+	status, err := strconv.Atoi(c.Query("taskStatus"))
 	// verify status is valid
 	if err != nil || (status != 0 && status != 1 && status != 2) {
 		fmt.Println("[PUT] incorrectly formatted status")
 		c.IndentedJSON(http.StatusBadRequest, gin.H{"message": "status should be 0, 1, or 2"})
 		return
 	}
 
-	if !db.UpdateTask(uid, taskid, content, status, task_date) {
-		fmt.Println("[PUT] cant find task id")
-		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "task with id " + taskid + " not found"})
+	if err := db.UpdateTask(uid, taskid, content, status, task_date); err != nil {
+		fmt.Println("[PUT] failed to update")
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "failed to update " + taskid + " in db (maybe wrong id?)"})
+		return
+	}
+	task, err := db.GetUserTaskByID(uid, taskid)
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "error getting updated task " + taskid})
 		return
 	}
-	c.IndentedJSON(http.StatusOK, db.SelectUserTaskByID(uid, taskid))
+	c.IndentedJSON(http.StatusOK, task)
 }
 
 func DeleteTask(c *gin.Context) {
 	uid, taskid := c.Query("uid"), c.Query("taskID")
-	task := db.SelectUserTaskByID(uid, taskid)
-	if !db.DeleteTask(uid, taskid) {
-		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "task with id " + taskid + " not found"})
+	task, err := db.GetUserTaskByID(uid, taskid)
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "error getting task to delete " + taskid})
+		return
+	}
+	if err := db.DeleteTask(uid, taskid); err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "failed to delete " + taskid + " in db (maybe wrong id?)"})
```

Comment:
do enums exist in go? if so, export an enum with all error messages. _We don't like magic strings_

Reply:
will do in logging ticket 🫡